contentgrid-appserver-query-engine-impl-jooq/src/main/java/com/contentgrid/appserver/query/engine/jooq/JOOQQueryEngine.java

```diff
@@ -422,6 +440,9 @@ public Optional<EntityData> delete(@NonNull Application application, @NonNull En
                     strategy.delete(dslContext, application, relation, entityRequest.getEntityId());
                 }
             }
+
+            deleteEventConsumer.onEntityDelete(application, data);
```

Comment:
You are emitting an event here *before* the actual delete has happened.

The event should be emitted after the delete has been done, but before returning from the function.